src/box/memtx_engine.cc

```diff
@@ -295,6 +384,43 @@ memtx_engine_recover_snapshot(struct memtx_engine *memtx,
 	if (xlog_cursor_open(&cursor, filename) < 0)
 		return -1;
 
+	/*
+	 * If Tarantool has started in force_recovery mode, we insert tuples
+	 * into SKs one by one to detect unique key violations, so bulk load
+	 * of secondary keys with O(n) sort is not applicable. Let's ignore
+	 * the sort data file in this case.
+	 *
+	 * If the `_index` space has a before_replace trigger, it might change
+	 * the contents of the space thus affecting the indexes we are about
+	 * to recover. So the sort data we've saved for a specific index_def
+	 * might get invalid once the index had been changed using a trigger.
+	 * E. g.: we've saved the sort data of an index with hints enabled,
+	 * but on recovery we made the index hintless using the before_replace
+	 * trigger on the `_index` space. Now the saved sort data is invalid
+	 * for the altered index.
+	 */
+	if (memtx->sort_data_enabled) {
```

Comment:
Better keep these checks together with the code that loads sort data.

Reply:
The `memtx_sort_data.c` file is unaware of memtx stuff, it only puts some data into and reads it from the file of the specified format. Seems better to keep memtx engine details out of the file.